packages/infra/lib/api-stack.ts

```diff
@@ -1080,6 +1095,64 @@ export class APIStack extends Stack {
     return documentDownloaderLambda;
   }
 
+  private setupBulkUrlSigningLambda(ownProps: {
+    lambdaLayers: LambdaLayers;
+    vpc: ec2.IVpc;
+    medicalDocumentsBucket: s3.Bucket;
+    fhirServerUrl: string;
+    envType: EnvType;
+    sentryDsn: string | undefined;
+    alarmAction: SnsAction | undefined;
+    searchEndpoint: string;
+    searchIndex: string;
+    searchUserName: string;
+    searchPassword: string;
+    apiService: ecs_patterns.NetworkLoadBalancedFargateService;
+  }): Lambda {
+    const {
+      lambdaLayers,
+      vpc,
+      medicalDocumentsBucket,
+      fhirServerUrl,
+      sentryDsn,
+      alarmAction,
+      envType,
+      searchEndpoint,
+      searchIndex,
+      searchUserName,
+      searchPassword,
+      apiService,
+    } = ownProps;
+
+    const bulkUrlSigningLambda = createLambda({
+      stack: this,
+      name: "BulkUrlSigning",
+      runtime: lambda.Runtime.NODEJS_18_X,
+      entry: "document-bulk-signer",
+      envType,
+      envVars: {
+        MEDICAL_DOCUMENTS_BUCKET_NAME: medicalDocumentsBucket.bucketName,
+        FHIR_SERVER_URL: fhirServerUrl,
+        SEARCH_ENDPOINT: searchEndpoint,
+        SEARCH_INDEX: searchIndex,
+        SEARCH_USERNAME: searchUserName,
+        SEARCH_PASSWORD: searchPassword,
+        API_URL: `http://${apiService.loadBalancer.loadBalancerDnsName}`,
+        ...(sentryDsn ? { SENTRY_DSN: sentryDsn } : {}),
+      },
+      layers: [lambdaLayers.shared],
+      memory: 512,
+      timeout: Duration.minutes(5),
```

Comment:
Since we have WHs and all, I'd have this close to 15min (max) so we minimize ops work and issues if/when we get a lot of records.